File: app/src/main/java/com/mhacks/android/ui/map/LocationManager.java
package com.mhacks.android.ui.map;

import com.mhacks.android.data.model.Location;
import com.mhacks.android.data.network.HackathonCallback;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Omkar Moghe on 10/7/2016.
 */

public class LocationManager {
    private ArrayList<Location> locations;
    private static LocationManager instance;

    public static LocationManager getInstance() {
        if (instance == null) {
            instance = new LocationManager();
        }

        return instance;
    }

    private LocationManager () {
//        NetworkManager networkManager = NetworkManager.getInstance();
//        networkManager.getLocations(new HackathonCallback<List<Location>>() {
//            @Override
//            public void success(List<Location> response) {
//                locations = new ArrayList<Location>(response);
//            }
//
//            @Override
//            public void failure(Throwable error) {
//                locations = new ArrayList<Location>();
//            }
//        });
    }

    public static Location getLocation(String id) {
        LocationManager manager = LocationManager.getInstance();

        for (Location l : manager.locations) {
            if (l.getId().equals(id)) return l;
        }

        return null;
    }
}

File: app/src/main/kotlin/com/mhacks/android/ui/kotlin/map/LocationManager.kt
package com.mhacks.android.ui.kotlin.map

import com.mhacks.android.data.model.Location
import com.mhacks.android.data.network.HackathonCallback
import com.mhacks.android.data.network.NetworkManager
import java.util.ArrayList

/**
 * Created by Shashank on 7/15/2017.
 */

class LocationManager private constructor() {
   /* private lateinit var locations: ArrayList<Location>

    init {
        val networkManager = NetworkManager.getInstance()
        networkManager.getLocations(object : HackathonCallback<List<Location>> {
            override fun success(response: List<Location>) {
                locations = ArrayList(response)
            }

            override fun failure(error: Throwable) {
                locations = ArrayList<Location>()
            }
        })
    }

    companion object {
        private lateinit var instance: LocationManager

        fun getInstance(): LocationManager {
            if (instance == null) {
                instance = LocationManager()
            }

            return instance
        }

        fun getLocation(id: String): Location? {
            val manager = LocationManager.getInstance()

            for (l in manager.locations!!) {
                if (l.getId() == id) return l
            }

            return null
        }
    }*/
}
